api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ViralLoadIntensiveMonitoringCohortQueries.java

```diff
@@ -0,0 +1,2685 @@
+            + "        GROUP BY p.patient_id "
+            + "     )min_vl "
+            + " GROUP BY min_vl.patient_id) final_min ON final_min.patient_id = p.patient_id "
+            + " WHERE p.patient_id NOT IN ( "
+            + "                            SELECT p.patient_id "
+            + "                            FROM patient p "
+            + "                                     INNER JOIN encounter e ON e.patient_id = p.patient_id "
+            + "                                     INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+            + "                            WHERE p.voided = 0 "
+            + "                              AND o.voided = 0 "
+            + "                              AND e.voided = 0 "
+            + "                              AND e.encounter_type = ${53} "
+            + "                              AND o.concept_id IN (${1982},${6332}) "
+            + "                              AND o.value_coded = ${1065} "
+            + "                              AND o.obs_datetime = final_min.final_min_date "
+            + "                            UNION "
+            + "                            SELECT p.patient_id "
+            + "                            FROM patient p "
+            + "                                     INNER JOIN encounter e ON e.patient_id = p.patient_id "
+            + "                                     INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+            + "                            WHERE p.voided = 0 "
+            + "                              AND o.voided = 0 "
+            + "                              AND e.voided = 0 "
+            + "                              AND e.encounter_type = ${6} "
+            + "                              AND o.concept_id IN (${1982},${6332}) "
+            + "                              AND o.value_coded = ${1065} "
+            + "                              AND e.encounter_datetime = final_min.final_min_date)"
+            + " ) AS xquery ON xquery.patient_id=p.patient_id "
+            + " WHERE p.voided = 0 "
+            + "   AND e.voided = 0"
+            + "   AND e.encounter_type = ${35} "
+            + "   AND TIMESTAMPDIFF(DAY, xquery.final_min_date, e.encounter_datetime) BETWEEN 20 AND 33 "
+            + "   AND e.location_id = :location "
+            + " GROUP BY p.patient_id) AS second_apss on second_apss.patient_id= p.patient_id"
+            + "  WHERE p.voided = 0 "
+            + "                      AND e.voided = 0"
+            + "                       AND e.encounter_type = ${35} "
+            + "                        AND TIMESTAMPDIFF(DAY, second_apss.min_apss, e.encounter_datetime) BETWEEN 20 AND 33 "
+            + "                        AND e.location_id = :location ";
+
+    StringSubstitutor sb = new StringSubstitutor(map);
+    sqlCohortDefinition.setQuery(sb.replace(sql));
+    return sqlCohortDefinition;
+  }
+
```

Comment:
this is callled multiple times tn the code to  attend different scenarios  of the spec,  and we went that way because we wanted to avoid  code     repitiion